easybuild/easyblocks/l/lammps.py

```diff
@@ -96,6 +97,11 @@ class EB_LAMMPS(CMakeMake):
     Support for building and installing LAMMPS
     """
 
+    def __init__(self, *args, **kwargs):
+        super(EB_LAMMPS, self).__init__(*args, **kwargs)
+        self.cuda = 'cuda' in [dep['name'].lower() for dep in self.cfg.dependencies()] or \
+                    hasattr(self.toolchain, 'COMPILER_CUDA_FAMILY')
```

Comment:
What's wrong with just using get_software_root('CUDA') ?

Reply:
There is no way to turn off CUDA right now, in our case CUDA leaks in because it is a dep of UCX way down the stack, but a CUDA build will not run on a node without GPUs so we need a way be explicit about whether we want CUDA or not

Reply:
Indeed, this approach is better: only if we're building with CUDA included as a dep or with a CUDA-enabled toolchain should CUDA support be enabled.

It probably doesn't hurt to create an easy method for this in the `Toolchain` class in framework?